src/pydata_sphinx_theme/assets/styles/sections/_sidebar-primary.scss

```diff
@@ -140,61 +141,110 @@
+
+          .toctree-toggle {
+            // Prevent toggle icon from getting squished by summary being a
+            // flexbox
+            flex: 0 0 auto;
+
+            // Override the rule below that reduces the font size to 0.75rem
+            .fa-chevron-down {
+              font-size: 1rem;
+            }
+          }
+        }
       }
     }
   }
-  // Navigation item chevrons
-  label.toctree-toggle {
-    position: absolute;
-    top: 0;
-    right: 0;
-    height: 30px;
-    width: 30px;
-
-    cursor: pointer;
+  li.has-children {
+    $toctree-toggle-width: 30px;
 
-    display: flex;
-    justify-content: center;
-    align-items: center;
+    position: relative;
 
-    &:hover {
-      background: var(--pst-color-surface);
+    > .reference,
+    .caption {
+      margin-right: calc($toctree-toggle-width + $focus-ring-width);
+      padding-top: 0.25rem; // align caption text with toggle chevron
     }
 
-    i {
-      display: inline-block;
-      font-size: 0.75rem;
-      text-align: center;
-      &:hover {
-        color: var(--pst-color-primary);
+    > details {
+      > summary {
+        // Remove browser default toggle icon
+        list-style: none;
+        &::-webkit-details-marker {
+          display: none;
+        }
+
+        // The summary element is natively focussable, but delegate the focus
+        // state to the toggle icon
+        &:focus-visible {
+          outline: none;
+
+          > .toctree-toggle {
+            // Prevent right side of focus ring from disappearing underneath the sidebar's right edge
+            outline: $focus-ring-outline;
+            outline-offset: -$focus-ring-width;
+          }
+        }
+
+        // Container for expand/collapse chevron icon
+        .toctree-toggle {
+          cursor: pointer;
+
+          // Position it so that it's aligned with the top right corner of the
+          // last positioned element, in this case the li.has-children
+          position: absolute;
+          top: 0;
+          right: 0;
+
+          // Give it dimensions
+          width: $toctree-toggle-width;
+          height: $toctree-toggle-width; // make it square
+
+          // Vertically and horizontally center the icon within the container
+          display: inline-flex;
+          justify-content: center;
+          align-items: center;
+
+          .fa-chevron-down {
+            font-size: 0.75rem;
+          }
+        }
+      }
+
+      // The section is open/expanded, rotate the toggle icon (chevron) so it
+      // points up instead of down
+      &[open] {
+        > summary {
+          .fa-chevron-down {
+            transform: rotate(180deg);
+          }
+        }
       }
-    }
-  }
-  .label-parts {
-    width: 100%;
-    height: 100%;
-    &:hover {
-      background: none;
-    }
-    i {
-      width: 30px;
-      position: absolute;
-      top: 0.3em; // aligning chevron with text
-      right: 0em; // aligning chevron to the right
     }
```

Comment:
The other parts of this PR suggest that this class was applied to the `label` element when styling the toggle icon for a level 0 heading. It's purpose, I think was to make the entire heading clickable. I refactored all of this so that the heading and the icon is put inside a summary tag, which makes the whole heading and the icon clickable.